pico/tools/src/main/java/io/helidon/pico/tools/Item.java

```diff
@@ -0,0 +1,200 @@
+ * <p>
+ * Note that generally speaking the majority of use cases can use methods from {@link ModuleInfoDescriptor}
+ * instead of needing to use anything provided here.
+ */
+@Builder
+public interface Item {
+
+    /**
+     * The comments that proceed the item definition.
+     *
+     * @return pre-comments
+     */
+    @Singular
+    List<String> precomments();
+
+    /**
+     * The target class name, package name, or module name this item applies to.
+     *
+     * @return target
+     */
+    String target();
+
+    /**
+     * Returns true if this is a {@code requires} definition.
+     *
+     * @return true if this item is using requires
+     */
+    boolean requires();
+
+    /**
+     * Returns true if this is a {@code uses} definition.
+     *
+     * @return true if this item is using requires
+     */
+    boolean uses();
+
+    /**
+     * Returns true if this is using a {@code transitive} definition.
+     *
+     * @return true if this item is using transitive
+     */
+    boolean isTransitiveUsed();
+
+    /**
+     * Returns true if this is using a {@code static} definition.
+     *
+     * @return true if this item is using static
+     */
+    // see https://github.com/helidon-io/helidon/issues/5440
+    boolean isStaticUsed();
+
+    /**
+     * Returns true if this is using a {@code exports} definition.
+     *
+     * @return true if this item is using exports
+     */
+    boolean exports();
+
+    /**
+     * Returns true if this is using a {@code open} definition.
+     *
+     * @return true if this item is using opens
+     */
+    boolean opens();
+
+    /**
+     * Returns true if this is using a {@code provides} definition.
+     *
+     * @return true if this item is using provides
+     */
+    boolean provides();
+
+    /**
+     * Applicable if the target is referencing a list of target class or package names.
+     *
+     * @return ordering
+     */
+    Optional<ModuleInfoDescriptor.Ordering> ordering();
+
+    /**
+     * Any {@code with} or {@code to} definitions.
+     *
+     * @return the set of with or to definitions
+     */
+    @Singular
+    Set<String> withOrTo();
+
+    /**
+     * If there is any {@link #ordering()} establish requiring a sort then a new sorted set will be returned.
+     *
+     * @return optionally the sorted set, else it is the same as {@link #withOrTo()}
+     */
+    default Set<String> withOrToSorted() {
+        if (ordering().isPresent()
+                && ordering().get() == ModuleInfoDescriptor.Ordering.SORTED) {
+            return new TreeSet<>(withOrTo());
+        }
+
+        return withOrTo();
+    }
+
+    /**
+     * Provides the content of the description item appropriate to write out.
+     *
+     * @return The contents (source code body) for this descriptor item.
+     */
+    default String contents() {
+        StringBuilder builder = new StringBuilder();
+        boolean handled = false;
+        if (uses()) {
+            assert (!requires());
```

Comment:
If this should be an actual runtime check, do not use `assert`
The assertions make this less readable and I do not see a benefit in adding them.

Reply:
Follow-up task updated https://github.com/helidon-io/helidon/issues/5440